### PR TITLE
diagnostics: updates to severity_sort

### PR DESCRIPTION
**Pull request by gpanders**
Update the default severity_sort order so that when `severity_sort` is true, higher severities are displayed before lower severities (e.g. ERROR is displayed over WARN).

### CODE REVIEW

File: runtime/doc/diagnostic.txt [hunk 1]
```diff
@@ -239,7 +239,9 @@ config({opts}, {namespace})                          *vim.diagnostic.config()*
                                  • severity_sort: (default false) Sort
                                    diagnostics by severity. This affects the
                                    order in which signs and virtual text are
-                                   displayed. Options:
+                                   displayed. When true, higher severities are
+                                   displayed before lower severities (e.g.
+                                   ERROR is displayed before WARN). Options:
```
Comment (justinmk) [inline runtime/doc/diagnostic.txt:244]: is ERROR (defined as `1`) considered "higher" than WARN (defined as `2`) ? Does this also apply to the min/max filters? so `severity={min=ERROR}` _only_ shows things "more severe" (but lower ordinal value)?

Comment (gpanders) [reply runtime/doc/diagnostic.txt:246]: Yes that's right. The numbering of the enum items is counter-intuitive (IMO) but is uses to preserve ease of compatibility with the LSP diagnostic severity definitions.

A "higher" severity is a "worse" severity, so ERROR is "greater than" WARN even though their numerical values are the opposite.

**Comment by gpanders** [comment]
@mjlbach I implemented the severity sorting by line as we discussed on Matrix in https://github.com/gpanders/neovim/pull/15701/commits/1f7714763fc24acda60b043406b3502d6c224636, would you mind taking a look when you get a chance?

**Comment by gpanders** [comment]
Squeezing in a few other small code cleanup/refactors that don't merit their own PR.

**Review by mjlbach — APPROVED** [review]

**Pull request merged**